src/element/textElement.ts

```diff
@@ -714,32 +709,36 @@ export const getTextBindableContainerAtPosition = (
   return isTextBindableContainer(hitElement, false) ? hitElement : null;
 };
 
-export const isValidTextContainer = (element: ExcalidrawElement) => {
-  return (
-    element.type === "rectangle" ||
-    element.type === "ellipse" ||
-    element.type === "diamond" ||
-    isImageElement(element) ||
-    isArrowElement(element)
-  );
-};
+const VALID_CONTAINER_TYPES = new Set([
+  "rectangle",
+  "ellipse",
+  "diamond",
+  "image",
+  "arrow",
+]);
 
-export const computeContainerHeightForBoundText = (
-  container: NonDeletedExcalidrawElement,
-  boundTextElementHeight: number,
+export type ValidContainerType<T> = T extends Set<infer U> ? U : never;
```

Comment:
let's move this to `src/global.d.ts` and make into a utility type. And could be slightly improved:

```ts
type ExtractSetType<T extends Set<any>> = T extends Set<infer U> ? U : never;
```